trade-finance-manager-ui/server/types/view-models/fee-record-details-view-model.ts

```diff
@@ -0,0 +1,13 @@
+import { SortedAndFormattedCurrencyAndAmount } from './utilisation-report-reconciliation-for-report-view-model';
+
+export type FeeRecordDetailsCheckboxId = `feeRecordId-${number}`;
+
+export type FeeRecordDetailsViewModel = {
+  id: number;
+  facilityId: string;
+  exporter: string;
+  reportedFees: SortedAndFormattedCurrencyAndAmount;
+  reportedPayments: SortedAndFormattedCurrencyAndAmount;
+  checkboxId?: FeeRecordDetailsCheckboxId;
+  isChecked?: boolean;
```

Comment:
This view model is supposed to be specific to the `fee-record-details-table.njk` component, so any view model which is rendering a page with this component should use this view model. Rather than having two versions, I'd prefer to use a generic which matches the `enableSelectingFeeRecords` option that we pass to the component which then returns the two shapes you describe

Reply:
I decided to settle on just having two separate types - one that has `WithCheckboxes` and one that has `WithoutCheckboxes`

Reply:
Yeah I think this is an interesting case where if we were setting the `enableSelectingFeeRecords` on the view model we could set it up so that the fields are required if that is true and not required if that is false, but I do think it is good for that to live in the templates using the macro rather than the view model. As it is I like your solution 😄